src/ethereum_test_tools/eof/v1/__init__.py

```diff
@@ -0,0 +1,492 @@
+                ]
+            )
+
+        # Add headers
+        concurrent_kind_sections: List[Section] = []
+        for s in sections:
+            if len(concurrent_kind_sections) == 0:
+                concurrent_kind_sections.append(s)
+                continue
+            if s.kind == concurrent_kind_sections[0].kind:
+                concurrent_kind_sections.append(s)
+                continue
+            else:
+                c += create_sections_header(
+                    concurrent_kind_sections[0].kind, concurrent_kind_sections
+                )
+                concurrent_kind_sections = [s]
+
+        if len(concurrent_kind_sections) > 0:
+            c += create_sections_header(
+                concurrent_kind_sections[0].kind, concurrent_kind_sections
+            )
+
+        # Add header terminator
+        if self.custom_terminator is not None:
+            c += self.custom_terminator
+        else:
+            c += EOF_HEADER_TERMINATOR
+
+        if self.auto_sort_sections:
+            # Sort sections for the body
+            sections = (
+                [s for s in sections if s.kind == SectionKind.TYPE]
+                + [s for s in sections if s.kind == SectionKind.CODE]
+                + [s for s in sections if s.kind == SectionKind.DATA]
+                + [s for s in sections if s.kind == SectionKind.CONTAINER]
+                + [
+                    s
+                    for s in sections
+                    if s.kind
+                    not in [
+                        SectionKind.TYPE,
+                        SectionKind.CODE,
+                        SectionKind.DATA,
+                        SectionKind.CONTAINER,
+                    ]
+                ]
+            )
+
+        # Add section bodies
+        for s in sections:
+            c += code_to_bytes(s.data if s.data is not None else "0x")
+
+        # Add extra (garbage)
+        if self.extra is not None:
+            c += self.extra
+
+        return c
+
+
+class Initcode(Code):
+    """
+    Helper class used to generate initcode for the specified deployment code,
+    using EOF V1 container as init code.
+    """
+
+    name: str
+    """
+    Name used to identify the initcode.
+    """
+    init_container: Container
+    assembled_output: bytes
+
+    def __init__(
+        self,
+        deploy_container: Container,
+        name: str = "EOF V1 Initcode",
+    ):
+        """
+        Generate legacy initcode that inits a contract with the specified code.
+        The initcode can be padded to a specified length for testing purposes.
+        """
+        self.name = name
+        self.assembled_output = deploy_container.assemble()
+        self.init_container = Container(
+            sections=[
+                Section(
+                    kind=SectionKind.CODE,
+                    data=Op.CALLDATACOPY(0, 0, len(self.assembled_output))
+                    + Op.RETURN(0, len(self.assembled_output)),
```

Comment:
Some clarification how this should work, maybe it will help:

So in the simplest case `init_container` can be something like just
```
code: RETURNCONTRACT{0}(0, 0)
container: assembled_output
data: empty
```
and then `initcode` can be
```
code: CREATE3{0}(0, 0, 0, 0) STOP
container: init_container
data: empty
```


More complicated case can append data to `init_container`, by for example getting it from calldata:
`init_container`:
```
code: CALLDATACOPY(0, CALLDATASIZE) RETURNCONTRACT{0}(0, CALLDATASIZE)
container: assembled_output
data: empty
```
`initcode`:
```
code: CALLDATACOPY(0, CALLDATASIZE) CREATE3{0}(0, 0, 0, CALLDATASIZE) STOP
container: init_container
data: empty
```

Reply:
Thanks for the insights! @spencer-tb is helping me to rebase to main right now, which should fix the issues you mentioned on the discord, and I think it would be very helpful if we start collaborating on the same branch to start improving these python tests.